src/base/bittorrent/infohash.h

```diff
@@ -38,15 +43,44 @@ using SHA256Hash = Digest32<256>;
 
 namespace BitTorrent
 {
-    class InfoHash : public SHA1Hash
+    class InfoHash;
+
+    class TorrentID : public SHA1Hash
```

Comment:
IMO using `public Digest32<160>` would be more correct as `SHA1Hash` implies a hash computed by a specific algorithm (SHA1) which is not our intention here.